### PR TITLE
Avoid unnecessary config update checks

### PR DESCRIPTION
**Pull request by qian0817**
### What changes are proposed in this pull request?

Check sleep timer interval change when update conf

### Why are the changes needed?

We currently generate these logs every time we execute the updateConf command, even if the value has not been modified.
```
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Cluster Metrics Updater interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Lost Worker Detection interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Worker register stream session cleaner interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Block Integrity Check interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master TTL Check interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Lost Files Detection interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Replication Check interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Persistence Scheduler interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Persistence Checker interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Metrics Time Series interval supplier.
2023-04-24 11:37:27,445 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Lost Master Detection interval supplier.
2023-04-24 11:37:27,446 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Log Config Report Scheduling interval supplier.
2023-04-24 11:37:27,446 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Journal Space Monitor interval supplier.
2023-04-24 11:37:27,446 INFO  [config-hash-master-heartbeat-0](SleepingTimer.java:88) - update Master Throttle interval supplier.

```

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

**Comment by Jackson-Wang-7** [comment]
@jiacheliu3 @dbw9580 please take a look

**Review by maobaolong — APPROVED** [review]
LGTM, thanks for this fix. @bzheng888 would you please take another look?

File: core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java [hunk 1]
```diff
@@ -56,4 +57,22 @@ public long getRunLimit(long mPreviousTickedMs) {
...
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    CronExpressionIntervalSupplier that = (CronExpressionIntervalSupplier) o;
+    return mInterval == that.mInterval
+        && Objects.equals(mCron.getCronExpression(), that.mCron.getCronExpression());
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java:71]: need a null check on `that.mCron`. I think you can add a `Preconditions.notNull(cronExpression, "CronExpression is null")` in the constructor

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM with a small comment. Thanks!

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM with a small comment. Thanks!

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM with a small comment. Thanks!

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM with a small comment. Thanks!

File: core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java (hunk 1 above)
Comment (qian0817) [reply core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java:73]: fix.

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks!

File: core/common/src/main/java/alluxio/heartbeat/SleepingTimer.java [hunk 2]
```diff
@@ -84,7 +85,10 @@ public long tick() throws InterruptedException {
 
   @Override
   public void update() {
-    mIntervalSupplier = mIntervalSupplierSupplier.get();
-    mLogger.info("update {} interval supplier.", mThreadName);
+    SleepIntervalSupplier newSupplier = mIntervalSupplierSupplier.get();
+    if (Objects.equals(mIntervalSupplier, newSupplier)) {
+      mIntervalSupplier = newSupplier;
+      mLogger.info("update {} interval supplier.", mThreadName);
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/heartbeat/SleepingTimer.java:91]: I don't understand - what's the point in updating the supplier if they compare equal?

Comment (qian0817) [reply core/common/src/main/java/alluxio/heartbeat/SleepingTimer.java]: oh, I wrote the condition incorrectly

**Review by dbw9580 — APPROVED** [review]

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**